domain-models-runtime/src/main/java/org/folio/rest/impl/RamlsAPI.java

```diff
@@ -0,0 +1,138 @@
+            Future.succeededFuture(
+              GetRamlsResponse.respond200WithApplicationJson(ramls)
+            )
+          );
+        } else {
+          String okapiUrl = okapiHeaders.get(OKAPI_URL_HEADER);
+          String raml = getRamlByPath(path, okapiUrl);
+          if (raml != null) {
+            asyncResultHandler.handle(
+              Future.succeededFuture(GetRamlsResponse.respond200WithApplicationRamlYaml(raml))
+            );
+          } else {
+            String notFoundMessage = "RAML " + path + " not found";
+            asyncResultHandler.handle(
+              Future.succeededFuture(
+                GetRamlsResponse.respond404WithTextPlain(notFoundMessage)
+              )
+            );
+          }
+        }
+      } catch (Exception e) {
+        log.error(e.getMessage(), e);
+        asyncResultHandler.handle(
+          Future.succeededFuture(
+            GetRamlsResponse.respond500WithTextPlain(e.getMessage())
+          )
+        );
+      }
+    });
+  }
+
+  private static List<String> getRamlsList() {
+    URL ramlListUrl = RamlsAPI.class.getClassLoader().getResource(RAMLS_PATH + GenerateRunner.RAML_LIST);
+    try {
+      if (ramlListUrl.toURI().getScheme().equals(JAR)) {
+        return IOUtils.readLines(RamlsAPI.class.getClassLoader().getResourceAsStream(RAMLS_PATH + GenerateRunner.RAML_LIST), StandardCharsets.UTF_8);
+      } else {
+        return Files.readAllLines(Paths.get(ramlListUrl.toURI()), StandardCharsets.UTF_8);
+      }
+    } catch (URISyntaxException | IOException e) {
+      return new ArrayList<>();
+    }
+  }
+
+  private List<String> getRamls() {
+    return RAMLS;
+  }
+
+  private String getRamlByPath(String path, String okapiUrl) throws URISyntaxException, IOException {
+    URL ramlUrl = getClass().getClassLoader().getResource(RAMLS_PATH + path);
+    if (ramlUrl == null) {
+      return null;
+    }
+    if (ramlUrl.toURI().getScheme().equals(JAR)) {
+      InputStream is = getClass().getClassLoader().getResourceAsStream(RAMLS_PATH + path);
+      return replaceReferences(IOUtils.toString(is, StandardCharsets.UTF_8.name()), okapiUrl);
+    } else {
+      return replaceReferences(IOUtils.toString(new FileInputStream(ramlUrl.getFile()), StandardCharsets.UTF_8.name()), okapiUrl);
+    }
+  }
+
+  private String replaceReferences(String raml, String okapiUrl) {
+    Matcher matcher = INCLUDE_MATCH_PATTERN.matcher(raml);
+    StringBuffer sb = new StringBuffer(raml.length());
+    while (matcher.find()) {
+      String path = matcher.group(0);
+      if (path.contains(RAMLS_PATH)) {
+        path = path.substring(path.lastIndexOf(RAMLS_PATH) + RAMLS_PATH.length());
+      }
+      if (path.endsWith(RAML_EXT)) {
+        matcher.appendReplacement(sb, Matcher.quoteReplacement(okapiUrl + "/_/ramls?path=" + path));
+      } else {
+        matcher.appendReplacement(sb, Matcher.quoteReplacement(okapiUrl + "/_/jsonSchemas?path=" + path));
+      }
+    }
```

Comment:
Can you add a unit test for the code inside of this while loop?

Reply:
Sure can.